File: iOS9/SearchAPIDemo/SearchAPIDemo/ViewController.m
//
//  ViewController.m
//  SearchAPIDemo
//
//  Created by Xiangjian Meng on 15/10/9.
//  Copyright © 2015年 mengxiangjian. All rights reserved.
//

#import "ViewController.h"
#import "Person.h"
#import "FriendViewController.h"

@interface ViewController () <UITableViewDataSource,UITableViewDelegate>
{
    NSMutableArray *personArray;
}

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    personArray = [NSMutableArray new];
    
    Person *becky = [self personWithName:@"becky"
                                personId:@"1"
                                   image:@"becky"];
    [personArray addObject:becky];
    
    Person *ben = [self personWithName:@"ben"
                                personId:@"2"
                                   image:@"ben"];
    [personArray addObject:ben];
    
    Person *jane = [self personWithName:@"jane"
                                personId:@"3"
                                   image:@"jane"];
    [personArray addObject:jane];
    
    Person *pete = [self personWithName:@"pete"
                                personId:@"4"
                                   image:@"pete"];
    [personArray addObject:pete];
    
    Person *ray = [self personWithName:@"ray"
                                personId:@"5"
                                   image:@"ray"];
    [personArray addObject:ray];
    
    Person *tom = [self personWithName:@"tom"
                                personId:@"6"
                                   image:@"tom"];
    [personArray addObject:tom];
    
    UITableView *tableView = [[UITableView alloc] initWithFrame:CGRectMake(0, 64, self.view.bounds.size.width, self.view.bounds.size.height - 64)
                                                          style:UITableViewStylePlain];
    tableView.delegate = self;
    tableView.dataSource = self;
    [self.view addSubview:tableView];
}

- (Person *)personWithName:(NSString *)name
                  personId:(NSString *)personId
                     image:(NSString *)imageName
{
    Person *person = [Person new];
    person.name = name;
    person.personId = personId;
    person.image = [UIImage imageNamed:imageName];
    
    return person;
}

#pragma mark - delegate datasource
- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return [personArray count];
}
- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    static NSString *identifier = @"cell";
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:identifier];
    if (!cell)
    {
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault
                                      reuseIdentifier:identifier];
    }
    
    Person *person = personArray[indexPath.row];
    
    cell.textLabel.text = person.name;
    cell.imageView.image = person.image;
    
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
    Person *person = personArray[indexPath.row];
    
    [self showPersonWithPersonId:person.personId];
}

- (void)showPersonWithPersonId:(NSString *)personId
{
    for (Person *person in personArray)
    {
        if ([person.personId isEqualToString:personId])
        {
            FriendViewController *vc = [[FriendViewController alloc] initWithPerson:person];
            [self.navigationController pushViewController:vc animated:YES];
            break;
        }
    }
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: iOS9/ForceTouchDemo/ForceTouchDemo/ViewController.swift
//
//  ViewController.swift
//  ForceTouchDemo
//
//  Created by Xiangjian Meng on 15/10/21.
//  Copyright © 2015年 mengxiangjian. All rights reserved.
//

import UIKit

class ViewController: UIViewController, UITableViewDataSource,UITableViewDelegate, UIViewControllerPreviewingDelegate {
    
    let tableView = UITableView()
    let names = ["张磊","林溢泽","王立群","孟祥建"]

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        self.navigationController?.navigationBar.translucent = false;
        
        self.title = "Force Touch Demo"
        tableView.delegate = self;
        tableView.dataSource = self;
        tableView.frame = CGRect(x: 0, y: 0, width: view.bounds.size.width, height: view.bounds.size.height)
        view.addSubview(tableView)
        
    }
    
    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return names.count;
    }
    
    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        
        let identifier = "cell"
        var cell = tableView.dequeueReusableCellWithIdentifier(identifier)
        if cell == nil {
            cell = UITableViewCell(style: .Default, reuseIdentifier: identifier)
        }
        cell?.textLabel?.text = names[indexPath.row]
        
        if traitCollection.forceTouchCapability == UIForceTouchCapability.Available {
            registerForPreviewingWithDelegate(self, sourceView: cell!)
        }
        
        return cell!
    }
    
    // 预览delegate方法 第一个方法为预览时调用，返回预览的viewController
    func previewingContext(previewingContext: UIViewControllerPreviewing, viewControllerForLocation location: CGPoint) -> UIViewController? {
//        guard let indexPath = tableView.indexPathForRowAtPoint(location),
//            cell = tableView.cellForRowAtIndexPath(indexPath)
//            else {return nil}
        let cell = previewingContext.sourceView as! UITableViewCell
        
        let detailViewController = DetailViewController()
        detailViewController.text = cell.textLabel!.text!
        return detailViewController
    }
    
    // 继续force touch就会直接显示
    func previewingContext(previewingContext: UIViewControllerPreviewing, commitViewController viewControllerToCommit: UIViewController) {
        showViewController(viewControllerToCommit, sender: self)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

